azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java

```
@@ -0,0 +1,126 @@
+
+import java.util.Set;
+
+public class VMImageToImage implements Function<VMImage, Image> {
+
+   private static final String UNRECOGNIZED = "UNRECOGNIZED";
+
+   private static final String UBUNTU = "Ubuntu";
+
+   private static final String WINDOWS = "Windows";
+
+   private static final String OPENLOGIC = "openLogic";
+
+   private static final String CENTOS = "CentOS";
+
+   private static final String COREOS = "CoreOS";
+
+   private static final String OPENSUSE = "openSUSE";
+
+   private static final String SUSE = "SUSE";
+
+   private static final String SQL_SERVER = "SQL Server";
+
+   private static final String ORACLE_lINUX = "Oracle Linux";
+
+   private final Supplier<Set<? extends org.jclouds.domain.Location>> locations;
+
+   public static String encodeFieldsToUniqueId(VMImage imageReference){
+      return (imageReference.globallyAvailable ? "global" : imageReference.location) + "/" + imageReference.publisher + "/" + imageReference.offer + "/" + imageReference.sku;
+   }
+
+   public static String[] decodeFieldsFromUniqueId(final String id) {
+      final String[] parts = checkNotNull(id, "id").split("/");
+      return parts;
+   }
+
+   @Inject
+   VMImageToImage(@Memoized final Supplier<Set<? extends Location>> locations) {
+      this.locations = locations;
+   }
+
+   @Override
+   public Image apply(final VMImage image) {
+      FluentIterable.from(locations.get()).firstMatch(LocationPredicates.idEquals(image.location)).orNull();
+
+      final ImageBuilder builder = new ImageBuilder()
+              .name(image.offer)
+              .description(image.sku)
+              .status(Image.Status.AVAILABLE)
+              .version(image.sku)
+              .id(encodeFieldsToUniqueId(image))
+              .providerId(image.publisher)
+              .location(image.globallyAvailable ? null : FluentIterable.from(locations.get())
+                      .firstMatch(LocationPredicates.idEquals(image.location))
+                      .orNull());
```

Same comment about failing instead of returning null